clients/src/test/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegateTest.java

```diff
@@ -169,14 +180,38 @@ public void testHasAnyPendingRequests() throws Exception {
         }
     }
 
+    @Test
+    public void testPropagateMetadataError() {
+        AuthenticationException authException = new AuthenticationException("Test Auth Exception");
+        doThrow(authException).when(metadata).maybeThrowAnyException();
+
+        LinkedList<BackgroundEvent> backgroundEventQueue = new LinkedList<>();
+        this.backgroundEventHandler = new BackgroundEventHandler(backgroundEventQueue);
+        NetworkClientDelegate networkClientDelegate = newNetworkClientDelegate();
+
+        assertEquals(0, backgroundEventQueue.size());
+        networkClientDelegate.poll(0, time.milliseconds());
+        assertEquals(1, backgroundEventQueue.size());
+
+        BackgroundEvent event = backgroundEventQueue.poll();
+        assertNotNull(event);
+        assertEquals(BackgroundEvent.Type.ERROR, event.type());
+        assertEquals(authException, ((ErrorEvent) event).error());
```

Comment:
I see you created the authException above but we're not using it here to simplify, do you plan to?

Reply:
my bad, you got it (I was probably looking at an old commit?). All good